eel/edge.py

```diff
@@ -1,20 +1,48 @@
-import platform
+import os
 import subprocess as sps
 import sys
 from typing import List
 
 from eel.types import OptionsDictT
 
-name: str = 'Edge'
+name: str = "Edge"
 
 
-def run(_path: str, options: OptionsDictT, start_urls: List[str]) -> None:
-    cmd = 'start microsoft-edge:{}'.format(start_urls[0])
-    sps.Popen(cmd, stdout=sys.stdout, stderr=sys.stderr, stdin=sps.PIPE, shell=True)
+def run(path: str, options: OptionsDictT, start_urls: List[str]) -> None:
+    if path != "edge_legacy":
+        if options["app_mode"]:
+            for url in start_urls:
+                sps.Popen([path, "--app=%s" % url] + options["cmdline_args"], stdout=sps.PIPE, stderr=sps.PIPE, stdin=sps.PIPE)
+        else:
+            args = options["cmdline_args"] + start_urls
+            sps.Popen([path, "--new-window"] + args, stdout=sps.PIPE, stderr=sps.PIPE, stdin=sps.PIPE)
+    else:
+        cmd = "start microsoft-edge:{}".format(start_urls[0])
+        sps.Popen(cmd, stdout=sps.PIPE, stderr=sps.PIPE, stdin=sps.PIPE, shell=True)
 
 
 def find_path() -> bool:
```

Comment:
`find_path()` does not return a boolean anymore, so the type annotation must be changed or it won't pass the static typing tests. I believe this should return `Optional[str]` now.